### PR TITLE
Restrict to supported media types

### PR DESCRIPTION
**Pull request by LeoniePeters**

### CODE REVIEW

File: src/response.js [hunk 1]
```diff
@@ -52,20 +52,24 @@ const thumbnailUrl = (providerAggregation, width) => {
...
+const oEmbedType = ({ rights, mediaType }) => {
+  return embeddableMediaType(mediaType) && embeddableRights(rights) ? 'rich' : 'link';
+};
+
+const embeddableMediaType = (mediaType) => {
+  return (typeof mediaType === 'string') &&
+        ['audio', 'image', 'video'].includes(mediaType.split('/')[0]);
+};
 
+const embeddableRights = (rights) => {
```
Comment (lbiedinger) [inline src/response.js]: Rename to `embeddingPermitted`

File: src/response.js [hunk 2]
```diff
@@ -52,20 +52,24 @@ const thumbnailUrl = (providerAggregation, width) => {
   return url.toString();
 };
 
-const typeForRights = (rights) => {
-  if (!rights) return 'link';
+const oEmbedType = ({ rights, mediaType }) => {
+  return embeddableMediaType(mediaType) && embeddableRights(rights) ? 'rich' : 'link';
+};
+
+const embeddableMediaType = (mediaType) => {
```
Comment (lbiedinger) [inline src/response.js]: Rename to `mediaTypeEmbeddable`

File: src/response.js [hunk 3]
```diff
@@ -111,12 +115,50 @@ const oEmbedResponseForIdentifier = async(identifier, options = {}) => {
...
 const oEmbedResponseForItem = (item, options = {}) => {
   const europeanaProxy = item.proxies.find(proxy => proxy.europeanaProxy);
   const providerProxy = item.proxies.find(proxy => !proxy.europeanaProxy);
   const providerAggregation = item.aggregations[0];
-  const edmIsShownByWebResource = providerAggregation.webResources
-    .find(webResource => webResource.about === providerAggregation.edmIsShownBy);
+
+  const mediaUris = [providerAggregation.edmIsShownBy]
+    .concat(providerAggregation.hasView || [])
+    .filter(media => media !== undefined);
```
Comment (pixelomo) [inline src/response.js:156]: It might be better to use `typeof media !== 'undefined'`

Comment (rwd) [reply src/response.js:156]: Not necessary as this is a Node.js app.

File: src/response.js [hunk 4]
```diff
@@ -52,20 +52,24 @@ const thumbnailUrl = (providerAggregation, width) => {
...
 
-const typeForRights = (rights) => {
-  if (!rights) return 'link';
+const oEmbedType = ({ rights, mediaType }) => {
+  return mediaTypeEmbeddable(mediaType) && embeddingPermitted(rights) ? 'rich' : 'link';
+};
+
+const mediaTypeEmbeddable = (mediaType) => {
+  return (typeof mediaType === 'string') &&
+        ['audio', 'image', 'video'].includes(mediaType.split('/')[0]);
```
Comment (rwd) [inline src/response.js]: There is one additional case we need to support, where the full media type is `application/dash+xml`.

Please also include in a comment a link to the source logic for this: https://github.com/europeana/metis-framework/blob/v2.1/metis-common/src/main/java/eu/europeana/metis/utils/MediaType.java#L41-L64

**Review by rwd — CHANGES_REQUESTED** [review]

**Review by rwd — APPROVED** [review]

**Pull request merged**